chore: Fury header add language field (#1612)
apache/incubator-fury: docs/specification/xlang_serialization_spec.md

Fury header consists starts one byte:

```
|    2 bytes   |     4 bits    | 1 bit | 1 bit | 1 bit  | 1 bit |         1 byte           |          optional 4 bytes          |
+--------------+---------------+-------+-------+--------+-------+--------------------------+------------------------------------+

> | magic number | reserved bits |  oob  | xlang | endian | null  |  serialization language  | unsigned int for meta start offset |

Reviewer: `serialization language` looks a little weird, how about removing serialization

Author: Updated

```

- magic number: used to identify fury serialization protocol, current version use `0x62d4`.
- null flag: 1 when object is null, 0 otherwise. If an object is null, other bits won't be set.
- endian flag: 1 when data is encoded by little endian, 0 for big endian.
- xlang flag: 1 when serialization uses xlang format, 0 when serialization uses Fury java format.
- oob flag: 1 when passed `BufferCallback` is not null, 0 otherwise.

> - language: the language when serializing objects, such as JAVA, PYTHON, GO, etc.

Reviewer: Could we add a note that Fury can use this flag to determine whether spend more time on serialization to make the deserialization faster for dynamic languages.

Author: Done

> - serialization language: indicates the language used for serialization, such as JAVA, PYTHON, GO, etc.

Reviewer: How about:
```suggestion
- language: the language when serializing objects, such as JAVA, PYTHON, GO, etc.
```

Author: Updated
